### PR TITLE
Dockerfile, cmake: Simplify build

### PR DESCRIPTION
**Pull request by woodruffw**
Always use clang in the Dockerfile, and don't overspecify
the build type.

Additionally, drop -Weverything when building in Debug mode --
it has competing flags internally and isn't intended for actual
builds.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,15 +7,15 @@ LABEL creator "Trail of Bits"
 LABEL dockerfile_maintenance "William Woodruff <william@trailofbits>"
 LABEL desc "Principled, lightweight C/C++ PE parser"
 
-RUN apk add --no-cache cmake icu-dev build-base
+RUN apk add --no-cache cmake icu-dev clang build-base
```
Comment (ekilmer) [inline Dockerfile]: It seems we'll need to set
```
CC=/usr/bin/clang
CXX=/usr/bin/clang++
```
environment variables or something to have CMake use `clang` during compilation

Right now, CMake inside Docker reports:
```
#9 [5/5] RUN mkdir build &&     cd build &&     cmake -DCMAKE_BUILD_TYPE="R...
#9 0.208 -- The C compiler identification is GNU 9.2.0
#9 0.251 -- The CXX compiler identification is GNU 9.2.0
#9 0.257 -- Check for working C compiler: /usr/bin/cc
#9 0.300 -- Check for working C compiler: /usr/bin/cc -- works
#9 0.301 -- Detecting C compiler ABI info
#9 0.344 -- Detecting C compiler ABI info - done
#9 0.361 -- Detecting C compile features
#9 0.362 -- Detecting C compile features - done
#9 0.366 -- Check for working CXX compiler: /usr/bin/c++
#9 0.412 -- Check for working CXX compiler: /usr/bin/c++ -- works
#9 0.414 -- Detecting CXX compiler ABI info
#9 0.462 -- Detecting CXX compiler ABI info - done
#9 0.479 -- Detecting CXX compile features
#9 0.480 -- Detecting CXX compile features - done
```

Comment (woodruffw) [reply Dockerfile]: Blah, yeah. I'll fix that in a bit. Thanks for the catch!

**Comment by ekilmer** [comment]
Also, we don't seem to test the Docker image build in CI

**Comment by woodruffw** [comment]
> Also, we don't seem to test the Docker image build in CI

We don't, yeah. It's "tested" by the fact that we auto-build a container on Docker Hub for every commit, but there's nothing in the CI itself.

**Comment by ekilmer** [comment]
> we auto-build a container on Docker Hub for every commit

I did not know this! Okay, that's probably fine for now then, until we have a test suite. 👍

**Pull request merged**